include/tudocomp_stat/StatPhaseExtension.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <vector>
+#include <tudocomp_stat/json.hpp>
+
+namespace tdc {
+    using json = nlohmann::json;
+
+/// \brief Virtual interface for extensions.
+///
+/// When constructed, measurement shall start. During the extension's lifetime,
+/// multiple calls to write may occur.
+class StatPhaseExtension {
+public:
+    /// \brief Writes phase data.
+    /// \param data the data object to write to.
+    virtual void write(json& data) = 0;
+
+    /// \brief Propagates the data of a sub phase to this phase.
+    /// \param ext the corresponding extension in the sub phase.
+    virtual void propagate(const StatPhaseExtension& sub) = 0;
```

Comment:
Was bedeutet "propagation" hier genau? Kopieren alle Daten in die parentphase?

Reply:
Propagiere Daten _von_ der Unterphase `sub` zur aktuellen Phase (`this`).

Wird für monoton steigende Daten (wie z.B. STXXL-Stats) nicht gebraucht, sondern eher für Maximum-Reduktionen wie bei Speicher-Peaks als Beispiel.

Man muss in einer Implementierung einen Downcast von `sub` zum eigenen Typ durchführen, das ist so gedacht.